static/panes/executor.js

```diff
@@ -804,21 +806,27 @@ Executor.prototype.initCallbacks = function () {
 
     var optionsChange = _.debounce(
         _.bind(function (e) {
-            this.onOptionsChange($(e.target).val());
+            if (this.settings.executorCompileOnChange) {
+                this.onOptionsChange($(e.target).val());
```

Comment:
I'm a bit concerned about this - a "should we compile on change" flag is preventing an "onOptionsChange()" going through. I think the logic should live in the onOptionsChange, not in the event handler. Can discuss, but I'd think I could add logic to `onOptionsChange` and handle other things (like UI invalidation?) without a logic flag like "compile on change" affecting it.